Library/Homebrew/utils/user.rb

```diff
@@ -16,6 +16,9 @@ def gui?
   end
 
   def self.current
-    @current ||= new(Etc.getpwuid(Process.euid).name)
```

Comment:
The original version of the code was structured not only so we didn't have instantiate `User` more than once, but also so that we didn't have to call `Etc.getpwuid` more than once. 😄 How about restructuring this so we just return `@current` if it exists? For example, adding the line

`return @current if defined?(@current)`

Reply:
Sorbet does not have a problem with the return type of `@current` but rather the return value of `Etc.getpwuid` (it should never be nil). 

IIUC, `return @current if defined?(@current)` would not solve Sorbet's problem with `Etc.getpwuid`. What do you think?

Reply:
Sorry, I was unclear. 😄 I'm suggesting doing both. `return @current if defined?(@current)` will ensure that we return the existing object without calling code unnecessarily, but your other change is still needed.